.buildkite/steps/build-debian-packages.sh

```diff
@@ -39,5 +36,5 @@ for ARCH in "amd64" "386" "arm" "armhf" "arm64"; do
   chmod +x "$BINARY"
 
   # Build the debian package using the architectre and binary, they are saved to deb/
-  ./scripts/build-debian-package.sh "$ARCH" "$BINARY" "$AGENT_VERSION" "$BUILD_VERSION"
+  .buildkite/steps/ruby-env ./scripts/build-debian-package.sh "$ARCH" "$BINARY" "$AGENT_VERSION" "$BUILD_VERSION"
```

Comment:
leverage the new `.buildkite/steps/ruby-env` (also added in this PR) to launch a process inside a container that has ruby. 

This script will come in handy as other steps in the build (like building the rpm) are also converted to be runnable on elastic stacks